src/Symfony/Component/HttpKernel/Kernel.php

```diff
@@ -440,7 +442,12 @@ protected function build(ContainerBuilder $container)
      */
     protected function getContainerClass()
     {
-        return str_replace('\\', '_', \get_class($this)).ucfirst($this->environment).($this->debug ? 'Debug' : '').'Container';
+        $name = \get_class($this);
+        if (false !== $i = strrpos($name, '\\')) {
+            $name = substr($name, 1 + $i);
+        }
+
+        return $this->name.$name.ucfirst($this->environment).($this->debug ? 'Debug' : '').'Container';
```

Comment:
adding `$this->name` back is what fixes deps=high tests - to be removed in 5.0 (alongside with the property)